client/js/models/archive.js

```diff
@@ -191,7 +191,7 @@ module.exports = {
           if (file && file.progressListener && file.progressHandler) {
             file.progressListener.removeListener('progress', file.progressHandler)
           }
-          if (file.fullPath === '/dat.json') send('archive:updateMetadata', {}, noop)
+          if (file.fullPath === '/dat.json' || file.fullPath === '/datapackage.json') send('archive:updateMetadata', {}, noop)
```

Comment:
Also if there are both, we should prefer dat.json over datapackage.json, which might not come in that order from the list() callback.